### ISSUE TITLE
Complicance to ANSI Common Lisp

### ISSUE THREAD

**Issue by jcguu95** [cxxxr/valtan#18]
Does valtan compiles the full common lisp into javascript? (In this case, valtan implements another compliant common lisp!)

Or does it only compile a subset of common lisp? If so, is the precise subset specified somewhere?

**Comment by cxxxr** [comment]
Valtan is a implementation that aims to be compatible with ANSI Common Lisp.
The degree of compatibility with ANSI Common Lisp is tested using sacla-tests.
Please refer to the README.

**Comment by jcguu95** [comment]
Ah, I see. That's what you meant in README. Why don't use the [`ansi-test`](https://gitlab.common-lisp.net/ansi-test/ansi-test)? It seems to be much more complete than `sacla-tests`, and it is used by other full many other common lisp implementations as well.

**Comment by cxxxr** [comment]
I use sacla-tests because it seems easy to use.
Actually, ansi-tests might be better.

**Comment by jcguu95** [comment]
I see. If the objective is to achieve full compliance with ANSI Common Lisp, it seems there may still be a considerable amount of work to be done. It seems almost necessary to pass most of the de facto [ANSI test suite](https://gitlab.common-lisp.net/ansi-test/ansi-test), as it is widely regarded as the standard test suite used by many mainstream Common Lisp implementations:

+ sbcl: [sbcl/ansi-test](https://github.com/sbcl/ansi-test).
+ ccl: [ccl/lib/compile-ccl.lisp · Clozure/ccl](https://github.com/Clozure/ccl/blob/7e9d7c76f304c58e2f0b91d43d0289a5150a2cf0/lib/compile-ccl.lisp#L822).
+ clasp: [clasp/repos.sexp /clasp](https://github.com/clasp-developers/clasp/blob/08cf7cf661bc53e7cec88283c87739ec7e99b250/repos.sexp#L8), [clasp-developers/ansi-test](https://github.com/clasp-developers/ansi-test).
+ ecl: [src/tests/Makefile.in / ECL · GitLab](https://gitlab.com/embeddable-common-lisp/ecl/-/blob/develop/src/tests/Makefile.in#L22), [src/tests/config.lsp.in / ECL · GitLab](https://gitlab.com/embeddable-common-lisp/ecl/-/blob/develop/src/tests/config.lsp.in#L138).

I am excited to see this project succeed in that direction. I attempted to run valtan against the ANSI test suite, but it seems there is still some work to be done before it can successfully run the tests. I have made a [pull request](https://github.com/cxxxr/valtan/pull/19) to address this. While this might make Valtan seem less promising at the moment, I believe it would greatly benefit those of us, like myself, who are keen to assess how complete Valtan is as an ANSI Common Lisp implementation and might otherwise spend considerable time evaluating it.

### Attempts to run valtan against ansi-test.

Git clone [ANSI test suite](https://gitlab.common-lisp.net/ansi-test/ansi-test) into `/tmp/`. Edit `valtan/tests/main.lisp` as follows, and `cd valtan/tests; npm i; npm start;`.

```
;(run-sacla-tests)
(setf *compile-file-truename* nil)
      *load-truename* nil
      *default-pathname-defaults* #P"/private/tmp/ansi-test/"
(load "/tmp/ansi-test/doit.lsp")
```

One could tell that several things are still missing. For example, it does not support the standard syntax `#P`, does not implement the function `PATHNAME`, and it seems to not have implemented pathnames. It also does not have the function `COMPILE-FILE-PATHNAME`.

**Comment by cxxxr** [comment]
Yes, in that respect, sacla-tests was suitable for this phase because it could be incorporated from the stage when nothing had been implemented.
